src/coreclr/src/vm/assemblyloadcontext.cpp

```diff
@@ -14,3 +15,49 @@ HRESULT AssemblyLoadContext::GetBinderID(
     *pBinderId = reinterpret_cast<UINT_PTR>(this);
     return S_OK;
 }
+
+NativeImage *AssemblyLoadContext::LoadNativeImage(Module *componentModule, LPCUTF8 nativeImageName)
+{
+    STANDARD_VM_CONTRACT;
+
+#ifndef DACCESS_COMPILE
+    BaseDomain::LoadLockHolder lock(AppDomain::GetCurrentDomain());
+    AssemblyLoadContext *loadContext = componentModule->GetFile()->GetAssemblyLoadContext();
+    PTR_LoaderAllocator moduleLoaderAllocator = componentModule->GetLoaderAllocator();
+
+    int nativeImageCount = m_nativeImages.GetCount();
+    for (int nativeImageIndex = 0; nativeImageIndex < nativeImageCount; nativeImageIndex++)
+    {
+        NativeImage *nativeImage = m_nativeImages[nativeImageIndex];
+        if (nativeImage->Matches(nativeImageName, loadContext))
```

Comment:
We should only have native images from the right load context here, so it should not be necessary to test loadContext for the match.

Reply:
Good point, removed.